main.py

```diff
@@ -245,6 +274,148 @@ async def disable_rolling(self):
             logger.exception("Failed to delete rolling recording buffer files")
         logger.info("Disable rolling was called end")
 
+    async def create_decky_pa_sink(self):
+        logger.info("Making audio pipeline")
+        # Creates audio pipeline
+        audio_device_output = get_cmd_output("pactl get-default-sink")
+        # expected output: alsa_output.pci-0000_04_00.5-platform-acp5x_mach.0.HiFi__hw_acp5x_1__sink when using internal speaker
+        # bluez_output.20_74_CF_F1_C0_1E.1 when using bluetooth
+        logger.info(f"Creating {self._deckySinkModuleName}")
+
+        get_cmd_output(f"pactl load-module module-null-sink sink_name={self._deckySinkModuleName}")
+
+        get_cmd_output(f"pactl load-module module-loopback source={audio_device_output}.monitor sink={self._deckySinkModuleName}")
+
+        if await Plugin.is_mic_enabled(self):
+            await Plugin.attach_mic(self)
+
+    async def cleanup_decky_pa_sink(self):
+        unload_pa_modules("Echo-Cancelled")
+        unload_pa_modules(f"{self._deckySinkModuleName}")
+
+    async def get_default_mic(self):
+        return get_cmd_output("pactl get-default-source")
+
+    async def is_mic_enabled(self):
+        logger.info(f"Is mic enabled? {self._micEnabled}")
+        return self._micEnabled
+
+    async def is_mic_attached(self):
+        is_attached = subprocess.run("pactl list modules | grep 'Echo-Cancelled'",shell=True).returncode == 0
+        logger.info(f"Is mic attached? {is_attached}")
+        return is_attached
+
+    async def attach_mic(self):
+        logger.info(f"Attaching Microphone {self._echoCancelledMicName}")
+
+        # Download gpl-3 binary for superior denoising (cannot include in this repo as gpl)
+
+        # Define the URLs and file paths
+        denoise_version = "1.03"
```

Comment:
The canonical way of doing this in this repo is to download the necessary software during the backend build and copy it over to the plugin and then modify the RPATH so that the plugin can find the .so file. This also ensures that every user has a consistent experience.

Please refactor this part to do the setup for denoising in the backend. Alternatively you can remove the denoising parts from this PR and we can get it merged, and then you can open a new PR with the denoising stuff added using the backend (that is how I would do things at my job).

Reply:
Hi @safijari, I've decided to make the denoising binary optional and add instructions on how to install it in the readme. If will default to builtin pulseaudio noise cancellation (which is still pretty good) if the binary does not exist. I've verified that this does work through tailing the plugin logs and testing some recordings.

I think it will also be easier this way as to get the plugin changes updated in the Decky main plugins repo, as we don't need to check True to any of the questions related to 3rd party binaries. We can still try to include it in the future, but I don't have too much time until next weekend. Let me know what you think.